src/UI/SlotArea.cpp

```diff
@@ -1482,7 +1482,17 @@ void cSlotAreaEnchanting::Clicked(cPlayer & a_Player, int a_SlotNum, eClickActio
 		bAsync = true;
 	}
 	cItem & DraggingItem = a_Player.GetDraggingItem();
-
+	if (a_SlotNum == 1)
+	{
+		// Lapis slot can have a full stack handle it normally
+		if (DraggingItem.m_ItemType != E_BLOCK_LAPIS_ORE)
```

Comment:
This is lapis ore rather than harvested lapis.  For the item it should be `m_ItemType == E_ITEM_DYE` and `m_ItemDamage == E_META_DYE_BLUE`.

Reply:
Oops yep! Good catch.